src/igor/schedule.go

```diff
@@ -11,6 +11,14 @@ import (
 	"time"
 )
 
+func checkValidNodeRange(nodes []string) bool {
+	indexes, err := getNodeIndexes(nodes)
+	if err != nil {
+		log.Fatal("Unable to get node indexes: %v", err)
```

Comment:
We should avoid calling `log.Fatal` and instead return false here.

Most of the time, we will want to call `log.Fatal` if the node check fails but we don't want to not use the function because it could cause the program to exit. It will also make it easier to write a couple of test cases ;) ;).

Reply:
Because we are no longer returning err and just returning a generic boolean, I think having something that will more easily distinguish why we failed here will be helpful.